src/xcode/ENA/ENA/Source/Workers/Store/Store.swift

```diff
@@ -60,8 +60,15 @@ protocol StoreProtocol: AnyObject {
 
 	var riskCalculationResult: RiskCalculationResult? { get set }
 
+	/// Date when the risk was changed to high
 	var dateOfConversionToHighRisk: Date? { get set }
 	
+	/// Date when the test was registered
+	var testRegistrationDate: Date? { get set }
+	
+	/// Date when the test result came
+	var testResultDate: Date? { get set }
+
```

Comment:
The **testResultDate** and **testRegistrationDate** are only part of the **submittionMetadata** could you move them inside the mode?, or else future developers will get confused which one refers to what as we have the same properties for the testResultMetadata

Reply:
No, I need them to be separated as it can be used by other developers too. These values are true for both submission methods i.e., TAN and QR unlike what you did in your task which was QR specific. Key Submission Meta data model only represents the values which are a part of the Protobuf and doesn't contain any extra attributes.